### DISCUSSION TITLE
What does "Failed to read existing file" exactly means?

### DISCUSSION THREAD

**Discussion by thommycgn** [jomjol/AI-on-the-edge-device#1336] [General] [answered]
I use AI on the edge to read the water meter and the gas meter. Both in the basement, on a 2A power supply. I kept having problems with both devices because they suddenly showed the error "Failed to read existing file". I have updated both devices to the latest firmware version. The water meter has not caused any problems so far. But the gas meter is no longer usable. Again and again "Failed to read existing file". Unplug and reboot helps until the next reboot. Sometimes you have to try it 3 times until it works again.

I have already done the following: 
- Changed SD card
- Changed the power supply
- ESP connected directly to a power supply
- erased the ESP and reinstalled the newst firmware (12.0.1) with the ESP tool
- deleted all data on the SD card and downloaded and installed the data from Github
- cleaned camera contacts

What does the error message mean exactly? What could be the cause of the problem? Maybe the WLAN, power supply, the filesystem or  camera?

**Comment by caco3** [comment]
Most obvious thing would be the SD-Card, but you replaced it already.
The Firmware itself does not contain files, only the SD-Card. Did you check the configuration? Maybe it tries to read a `tflite` file which no longer exists?
The logfile would help us to trace it down!

Also it is possible that your ESP board went bad.

**Comment by thommycgn** [comment]
I started with a new installation.. so why should it read an old tflite file? I don*t think so.
I changed the board a few hours ago. No problems yet. Maybe some defect on the board... 
I will test it out with other software in the next days.

**Comment by thommycgn** [comment]
Today the same: "Failed to read existing file" on the gas meter.... 

I have already replaced everything except the camera because it was already set correctly and I wanted to save myself the trouble of setting it up. So today I will replace the camera. That would be the last idea I would have.

Or does anyone have another idea what could be going on? It would be nice to get more detailed error codes from the software so you know where the problem is as well.

**Reply by caco3** [reply]
so why do you not first provide the logfile as already requested?
Since my glas bowl is broken it is really hard to look into your device...!

**Reply by thommycgn** [reply]
Because there was nothing interesting in the log file. I have activated the debug mode yesterday to get more info.

Attached is the logfile from today. It has a failure at 4:22:17. At 10:23:27 I pulled the plug and restarted. There is no error in the logfile. Shortly before 12:00 I had 2 times again "Failed to read existing file". I had tried to download the logfile. It happened. There are again no error messages in the logfile. 

Slowly I have the idea, if it could not be caused by the cable length (4-5m) in the basement. That the ESP needs in certain situations particularly much energy, which he then perhaps just does not get. 2 times the download of the logfile went wrong. The files were smaller, not complete. I have now brought the ESP up to the office and connected it here. I could download the logfile completely... I will keep this in mind and try around some more.

Here the [Logfile](https://drive.google.com/file/d/1qgXrPArQEIkeXcgLf3-XEPaB1-0kbx85/view?usp=sharing).

**Reply by caco3** [reply]
So if "Failed to read existing file" is not from the logfile, where did you then see it?

It is hard to say what it is. But like @mad2xlc suggested, you could try the latest rolling.

**Reply by mad2xlc** [reply]
For me it was random in web interface. But after it is gone in current rolling. So i didn't create an issue on this.

**Reply by thommycgn** [reply]
On the Webinterface. White Screen with only "Failed to read existing file".

**Reply by caco3** [reply]
hmm...

Use the [Web based Console](https://jomjol.github.io/AI-on-the-edge-device/index.html) or Putty or so to get the log from USB. Hopefully this gives more information

**Reply by thommycgn** [reply]
The problem is that I have glued the ESP32 with hot glue, and every time I want to change something, I have to loosen everything first.
I just did that, went down to the basement with my laptop. Only I don't get the data read out via the serial port when I only connect RX & TX and still use the external power supply. It only works if I also connect VCC & GND to the laptop. 
If I disconnect the power source, then I no longer see the error message and would have to leave my laptop there until the error reappears. This is not possible. Or is there a trick?

**Reply by caco3** [reply]
You don't need to connect TX if you just want to get the log. (TX is only needed for flashing).
But you also have to connect GND to the laptop, else the lines are freely floating.

I also suggest to only run the laptop on battery to avoid different GND potential between laptop and the devices power supply.
This could destroy the device!

**Comment by mad2xlc** [comment]
Had this with the Version 12.0.1. Try a rolling version.

**Comment by DanielWe2** [comment]
Had also had the problem on all three of my ESP32 CAMs (two different vendors).

My solution:

- Reformat (and repartion) the SD Card (you can make a backup of all the files before you do that) to only have exactly one partition starting at the absoulte beginning.  Initially I had formated them using Windows which somehow put a small empty block at the beginning of the partional table.   The second time I created the partion with cfisk in Linux.   

- I check the power supply Voltage at the ESP itself.  The issue is:   When you use a longer but small diameter cable you can get quite a lot of voltage drop on the cable itself at the power peaks of the board.  In my case as low as 4.6 V I think that can cause issues.  I changed the power supply and am now using better cables.   A capacitor parallel to the power pins on the ESP board can also help to stabilize the voltage.

Not sure which one of those really made the difference but since then all of my 3 boards run stable without the "failed to read file issue".

**Reply by caco3** [reply]
@DanielWe2 Thanks for sharing your experience!

To my experience, long cables are not such an issue.
In my basement, I have 2 devices, they are both powered from the same USB Power supply (5V, 1.5A), one with 1m cable, the other with maybe 7m cable (0.25mm). And I never had issues, with power.

But indeed the partitioning is crucial. I also formatted with windows and did not have issue. But if there is a gap at the start or even a 2nd partition, it is likely that the ESP library can not handle this properly. But somehow it still can access them partially, else not Web Interface would be shown at all (its all stored on the SD-Card)...

**Comment by thommycgn** [comment, marked as answer]
In the meantime I could exclude that it is connected with the power supply.

**I think I have solved the problem now!**

I had specially ordered two 8 GB SD cards from Intenso for the ESP cam on eBay. After the problems I did read-write tests, which all passed. I swapped one Intenso card for the other Intenso card. Then I deleted the partitions, recreated them, and reformatted the cards. Still, the problems remained.

In the other ESP cam that I use for the water meter, I have a card from SanDisk in there, there have been no problems so far.

I have now replaced the SD card with an 8GB card from Netac. Suddenly everything runs through for days without problems!

In fact, **the problem seems to be with the SD cards from Intenso!** Why this is so, I do not know, however. When testing on the PC made the no problems. Maybe I was cheated on eBay garbage..

**Reply by caco3** [reply]
Danke für die Info!
Das müssen wir uns unbedingt merken, falls jemand wiede rmit diesem Fehler kommt.

Die Funktion zum Lesen/Schreiben der SD-Karte ist eine fertige Funktion von Espressif oder 3th Party. Kann gut sein, dass die nicht all zu intelligent ist.
